File: web/union-sql-inejection-mybatis/blog/src/main/java/com/example/blog/controller/PostController.java
package com.example.blog.controller;

import com.example.blog.domain.Post;
import com.example.blog.domain.PostSearchParam;
import com.example.blog.service.PostService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RequiredArgsConstructor
@RequestMapping("/api/v1/post")
@RestController
public class PostController {

    private final PostService postService;

    @GetMapping("")
    public List<Post> getPostsBySearchParam(PostSearchParam postSearchParam) {
        return postService.getPostsBySearchParam(postSearchParam);
    }
}

File: web/markdown-xss/blog/src/main/kotlin/org/example/blog/controller/PostController.kt
package org.example.blog.controller

import org.example.blog.domain.Post
import org.example.blog.service.PostService
import org.springframework.web.bind.annotation.*

@RequestMapping("/api/v1/post")
@RestController
class PostController(val postService: PostService) {
    @GetMapping("/{postId}")
    fun getPost(@PathVariable postId: Long): Post {
        return postService.getPost(postId)
    }

    @PostMapping("")
    fun createPost(@RequestBody post: Post): Post {
        return postService.createPost(post)
    }
}
